"RemovedCACertificateReportCSVFormat" file format has changed

Hi,

The number of columns in the CSV file "RemovedCACertificateReportCSVFormat" (https://ccadb-public.secure.force.com/mozilla/RemovedCACertificateReportCSVFormat) appears to have changed, and this is causing an error when executing the "inspect" command.

**internal/analyse/analyse.go**

```go
func downloadMozillaRemovedCACertsList() ([]removedCertificate, error) {
        const mozillaRemovedCACertificateReportURL = "https://ccadb-public.secure.force.com/mozilla/RemovedCACertificateReportCSVFormat"

        resp, err := http.Get(mozillaRemovedCACertificateReportURL)
        if err != nil {
                return nil, err
        }
        csvReader := csv.NewReader(resp.Body)
        csvLines, err := csvReader.ReadAll()
        if err != nil {
                return nil, err
        }

        removedCerts := make([]removedCertificate, len(csvLines))
        for i, csvLine := range csvLines {
                removedCerts[i] = removedCertificate{
                        // From the CSV format that Mozilla publishes, the 8th column (id 7) is the fingerprint and the 23rd column
                        // (id 22) is the comment.
                        Fingerprint: csvLine[7],
 ->                     Comments:    csvLine[22],
                }
        }
        return removedCerts, nil
}
```

```bash
$ paranoia inspect file://image.tar
panic: runtime error: index out of range [22] with length 21
goroutine 1 [running]:
github.com/jetstack/paranoia/internal/analyse.downloadMozillaRemovedCACertsList()
	/home/runner/work/paranoia/paranoia/internal/analyse/analyse.go:67 +0x2bf
github.com/jetstack/paranoia/internal/analyse.NewAnalyser(...)
	/home/runner/work/paranoia/paranoia/internal/analyse/analyse.go:41
github.com/jetstack/paranoia/cmd.newInspect.func2(0xc000136608?, {0xc00013a1c0?, 0x1?, 0x1?})
	/home/runner/work/paranoia/paranoia/cmd/inspect.go:49 +0xb6
github.com/spf13/cobra.(*Command).execute(0xc000136608, {0xc00013a180, 0x1, 0x1})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:985 +0xaaa
github.com/spf13/cobra.(*Command).ExecuteC(0xc000136008)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041
github.com/jetstack/paranoia/cmd.Execute()
	/home/runner/work/paranoia/paranoia/cmd/root.go:71 +0x1d
main.main()
	/home/runner/work/paranoia/paranoia/main.go:8 +0xf
```

The code is looking for column 23 to retrieve the comment, but that column no longer seems to exist in the new files. It appears the comment is now in column 21.

Instead of hard-coding the column index, maybe we should specify the name of the column to retrieve.

--- a/internal/analyse/analyse.go
+++ b/internal/analyse/analyse.go
@@ -16,8 +16,10 @@ import (
 type NoteLevel string
 
 const (
-	NoteLevelWarn  NoteLevel = "warn"
-	NoteLevelError NoteLevel = "error"
+	NoteLevelWarn     NoteLevel = "warn"
+	NoteLevelError    NoteLevel = "error"
+	fingerprintHeader           = "SHA-256 Fingerprint"
+	commentsHeader              = "Comments"
 )
 
 type Note struct {
@@ -53,45 +55,46 @@ func downloadMozillaRemovedCACertsList() ([]removedCertificate, error) {
 		return nil, err
 	}
 	csvReader := csv.NewReader(resp.Body)
-	csvLines, err := csvReader.ReadAll()
+	// Read the header first
+	headers, err := csvReader.Read()
 	if err != nil {
 		return nil, err
 	}
 
-	removedCerts := make([]removedCertificate, len(csvLines))
-	for i, csvLine := range csvLines {
-		// Skip the header row
-		if i == 0 {
-			continue
-		}
-
-		// Find column indices by their names from the header
-		var fingerprint, comments int
-		if i == 1 {
-			fingerprint = -1
-			comments = -1
-			for idx, header := range csvLines[0] {
-				switch header {
-				case "SHA-256 Fingerprint":
-					fingerprint = idx
-				case "Comments":
-					comments = idx
-				}
-			}
-
-			if fingerprint == -1 {
-				return nil, fmt.Errorf("required column 'SHA-256 Fingerprint' not found in CSV header")
-			}
-			if comments == -1 {
-				return nil, fmt.Errorf("required column 'Comments' not found in CSV header")
-			}
-		}
-
-		removedCerts[i-1] = removedCertificate{
-			Fingerprint: csvLine[fingerprint],
-			Comments:    csvLine[comments],
+	// Find column indices by their names from the header
+	fingerprint := -1
+	comments := -1
+	for idx, header := range headers {
+		switch header {
+		case fingerprintHeader:
+			fingerprint = idx
+		case commentsHeader:
+			comments = idx
 		}
 	}
+
+	if fingerprint == -1 {
+		return nil, fmt.Errorf("required column 'SHA-256 Fingerprint' not found in CSV header")
+	}
+	if comments == -1 {
+		return nil, fmt.Errorf("required column 'Comments' not found in CSV header")
+	}
+
+	// Now process each line individually instead of loading everything into memory
+	var removedCerts []removedCertificate
+	for {
+		record, err := csvReader.Read()
+		if err != nil {
+			break // End of file or other error
+		}
+
+		removedCerts = append(removedCerts, removedCertificate{
+			Fingerprint: record[fingerprint],
+			Comments:    record[comments],
+		})
+	}
+
+	// All CSV processing is done in the loop above, we already have the removedCerts slice
 	return removedCerts, nil
 }
 

--- a/internal/analyse/analyse.go
+++ b/internal/analyse/analyse.go
@@ -60,11 +60,36 @@ func downloadMozillaRemovedCACertsList() ([]removedCertificate, error) {
 
 	removedCerts := make([]removedCertificate, len(csvLines))
 	for i, csvLine := range csvLines {
-		removedCerts[i] = removedCertificate{
-			// From the CSV format that Mozilla publishes, the 8th column (id 7) is the fingerprint and the 23rd column
-			// (id 22) is the comment.
-			Fingerprint: csvLine[7],
-			Comments:    csvLine[22],
+		// Skip the header row
+		if i == 0 {
+			continue
+		}
+
+		// Find column indices by their names from the header
+		var fingerprint, comments int
+		if i == 1 {
+			fingerprint = -1
+			comments = -1
+			for idx, header := range csvLines[0] {
+				switch header {
+				case "SHA-256 Fingerprint":
+					fingerprint = idx
+				case "Comments":
+					comments = idx
+				}
+			}
+
+			if fingerprint == -1 {
+				return nil, fmt.Errorf("required column 'SHA-256 Fingerprint' not found in CSV header")
+			}
+			if comments == -1 {
+				return nil, fmt.Errorf("required column 'Comments' not found in CSV header")
+			}
+		}
+
+		removedCerts[i-1] = removedCertificate{
+			Fingerprint: csvLine[fingerprint],
+			Comments:    csvLine[comments],
 		}
 	}
 	return removedCerts, nil

